common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalShuffleBlockHandler.java

```diff
@@ -93,18 +113,34 @@ protected void handleMessage(
           client.getClientId(),
           NettyUtils.getRemoteAddress(client.getChannel()));
       callback.onSuccess(new StreamHandle(streamId, msg.blockIds.length).toByteBuffer());
+      transferBlockRate.mark(totalBlockSize / 1024 / 1024);
+      responseDelayContext.stop();
 
     } else if (msgObj instanceof RegisterExecutor) {
+      final Timer.Context responseDelayContext = timeDelayForRegisterExecutorRequest.time();
       RegisterExecutor msg = (RegisterExecutor) msgObj;
       checkAuth(client, msg.appId);
       blockManager.registerExecutor(msg.appId, msg.execId, msg.executorInfo);
       callback.onSuccess(ByteBuffer.wrap(new byte[0]));
+      responseDelayContext.stop();
 
     } else {
       throw new UnsupportedOperationException("Unexpected message: " + msgObj);
     }
   }
 
+  public MetricSet getAllMetrics() {
+    return metrics;
+  }
+
+  public long getRegisteredExecutorsSize() {
+    return blockManager.getRegisteredExecutorsSize();
+  }
+
+  public long getTotalShuffleRequests() {
+    return timeDelayForOpenBlockRequest.getCount() + timeDelayForOpenBlockRequest.getCount();
```

Comment:
Since there are only two metrics, I think it is better not to since it can be confusing. If there were many different kinds of requests maybe then a total would be ok.